### PR TITLE
fix: socket connection in RedisHandler class

### LINKED ISSUE #6929: Bug: Redis session with socket
### PHP Version

7.4

### CodeIgniter4 Version

4.2.10

### CodeIgniter4 Installation Method

Manual (zip or tar.gz)

### Which operating systems have you tested for this bug?

Linux

### Which server did you use?

apache

### Database

MariaDB 10.5.15

### What happened?

Codeigniter session throw this error when using Redis with socket:
**RedisException: php_network_getaddresses: getaddrinfo failed: Name or service not known.**

### Steps to Reproduce

Configure Codeigniter to use Redis session and use a socket host (ex: /tmp/redis.sock). 

### Expected Output

A working Redis session.

### Anything else?

In the RedisHandler session file, default port is set to 6379. When using a socket, we want a port set to 0. Here is my working patch on the system/Session/Handlers/RedisHandler.php file, line 121 :

change this : `if (! $redis->connect($this->savePath['host'], $this->savePath['port'], $this->savePath['timeout'])) {`
         
to this : `if (! $redis->connect($this->savePath['host'], ($this->savePath['host'][0] === '/' ? 0 : $this->savePath['port']), $this->savePath['timeout'])) {`

No more RedisException.

### PR DESCRIPTION
**Pull request by michalsn**
**Description**
Fixes #6929

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by ddevsr — CHANGES_REQUESTED** [review]
Remove this changes

File: user_guide_src/source/changelogs/v4.2.11.rst [hunk 1]
```diff
@@ -13,5 +13,7 @@ Bugs Fixed
 **********
 
 - Fixed a ``FileLocator::locateFile()`` bug where a similar namespace name could be replaced by another, causing a failure to find a file that exists.
+- Fixed a ``RedisHandler`` session class to use the correct config when used with a socket connection.
+- Fixed a ``FeatureTestTrait`` to use Services class for grabbing the ``IncomingRequest`` class.
```
Comment (ddevsr) [inline user_guide_src/source/changelogs/v4.2.11.rst:17]:
```suggestion
```

**Review by kenjis — CHANGES_REQUESTED** [review]
Please fix the user guide. Except it this PR looks good.

**Review by kenjis — APPROVED** [review]
LGTM!

**Comment by kenjis** [comment]
This fix seems to expect `$sessionSavePath` to be `/tmp/redis.sock`,
but should it be `unix:///var/run/redis/redis.sock?persistent=1&weight=1&database=0`?

See https://github.com/phpredis/phpredis?tab=readme-ov-file#php-session-handler
> phpredis can also connect to a unix domain socket: session.save_path = "unix:///var/run/redis/redis.sock?persistent=1&weight=1&database=0".

**Comment by kenjis** [comment]
And this fix was broken by #7187.
https://github.com/codeigniter4/CodeIgniter4/pull/7187/files#diff-1a7b1fe40ddc58f0b240a268889a1952778b0631ce9db5f54cddaa359429dd60R140

**Pull request merged**